### PR TITLE
Fix musl issue on container build

### PR DESCRIPTION
**Pull request by nikolaishields**
This PR does 2 things:

1. Build the go utilities with static flags
2. Updates the binary names to match the standard

### CODE REVIEW

**Review by briandowns — APPROVED** [review]

**Review by brooksn — APPROVED** [review]

File: cmd/gen_release_notes/Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ BINARY := gen_release_notes
 
 VERSION = v0.4.0
 GIT_SHA = $(shell git rev-parse HEAD)
-LDFLAGS = -ldflags "-X main.gitSHA=$(GIT_SHA) -X main.version=$(VERSION) -X main.name=$(BINARY)"
+LDFLAGS = -ldflags "-X main.gitSHA=$(GIT_SHA) -X main.version=$(VERSION) -X main.name=$(BINARY) -linkmode external -extldflags '-static -Wl,--fatal-warnings'"
```
Comment (wgsuse) [inline cmd/gen_release_notes/Makefile:8]: Like I mentioned before, This flag triggered a build warning and then it generated a binary that I was not able to execute.

```
[werner@gtm-lap-024 ~/code/ecm-distro-tools/cmd/gen-release-notes]$ rm -f bin/gen-release-notes && make                                          
CGO_ENABLED=1 go build -tags "netgo osusergo no_stage static_build" -v -ldflags "-X main.gitSHA=03b906ee378cc1d06778121117a39415c8b82bb7 -X main.version=v0.4.0 -X main.name=gen-release-notes -extldflags '-static' -linkmode external" -o bin/gen-release-notes
# github.com/rancher/ecm-distro-tools/cmd/gen-release-notes
loadinternal: cannot find runtime/cgo
[werner@gtm-lap-024 ~/code/ecm-distro-tools/cmd/gen-release-notes]$ ldd bin/gen-release-notes
        not a dynamic executable
[werner@gtm-lap-024 ~/code/ecm-distro-tools/cmd/gen-release-notes]$  bin/gen-release-notes -h
Trace/breakpoint trap
[werner@gtm-lap-024 ~/code/ecm-distro-tools/cmd/gen-release-notes]$
```

**Review by wgsuse — APPROVED** [review]
LGTM

File: cmd/gen_release_notes/Makefile (hunk 1 above)
Comment (nikolaishields) [reply cmd/gen_release_notes/Makefile:8]: Is this built against the code in the fork? The binary names don't line up.

Comment (wgsuse) [reply cmd/gen_release_notes/Makefile:8]: Yes, I checked out this PR branch and ran make on it:
```
werner@gtm-lap-024:~/code/ecm-distro-tools/cmd/gen_release_notes$ make
go clean
rm -f bin/gen_release_notes
CGO_ENABLED=1 go build -tags "netgo osusergo no_stage static_build" -v -ldflags "-X main.gitSHA=7165b23553ec5b31602b93aedb17ad75f8bd17e2 -X main.version=v0.4.0 -X main.name=gen_release_notes -linkmode external -extldflags '-static -Wl,--fatal-warnings'" -o bin/gen_release_notes
github.com/rancher/ecm-distro-tools/cmd/gen_release_notes
# github.com/rancher/ecm-distro-tools/cmd/gen_release_notes
loadinternal: cannot find runtime/cgo
werner@gtm-lap-024:~/code/ecm-distro-tools/cmd/gen_release_notes$ ls bin
gen_release_notes
werner@gtm-lap-024:~/code/ecm-distro-tools/cmd/gen_release_notes$ bin/gen_release_notes
Trace/breakpoint trap
133 werner@gtm-lap-024:~/code/ecm-distro-tools/cmd/gen_release_notes$
```

Comment (nikolaishields) [reply cmd/gen_release_notes/Makefile:8]: Could you build it in the container and see if the build errors are reproducible?

Comment (nikolaishields) [reply cmd/gen_release_notes/Makefile:8]: Thanks for clarifying, I'll attempt to reproduce.

File: cmd/backport/Makefile [hunk 2]
```diff
@@ -3,12 +3,13 @@ GO = go
...
 BINARY := backport
 
-VERSION = v0.1.0
-GIT_SHA = $(shell git rev-parse HEAD)
-LDFLAGS = -ldflags "-X main.gitSHA=$(GIT_SHA) -X main.version=$(VERSION) -X main.name=$(BINARY)"
-TAGS    = "netgo osusergo no_stage static_build"
+VERSION           = v0.1.0
+GIT_SHA           = $(shell git rev-parse HEAD)
+override LDFLAGS += -X main.gitSHA=$(GIT_SHA) -X main.version=$(VERSION) -X main.name=$(BINARY) -extldflags '-static -Wl,--fatal-warnings'
+# LDFLAGS = -ldflags "-X main.gitSHA=$(GIT_SHA) -X main.version=$(VERSION) -X main.name=$(BINARY)"
```
Comment (briandowns) [inline cmd/backport/Makefile:9]: Can we remove these from here and from the ones below?

**Pull request merged**